### PR TITLE
Add DAC support to STM32L1 series

### PR DESCRIPTION
**Pull request by tagunil**
Tested on a Nucleo-L152RE board. Both tests are passing, and the sample draws a nice sawtooth on my scope.
If that is good enough, I'll do the same for some more STM32 series.

### CODE REVIEW

File: drivers/pinmux/stm32/pinmux_stm32l1x.h [hunk 1]
```diff
@@ -214,4 +214,9 @@
 #define STM32L1X_PINMUX_FUNC_PF10_ADC1_IN31  \
 	STM32_MODER_ANALOG_MODE
 
+#define STM32L1X_PINMUX_FUNC_PA4_DAC_OUT1 \
```
Comment (gmarull) [inline drivers/pinmux/stm32/pinmux_stm32l1x.h:217]: pinmux should be placed in order.

File: boards/arm/nucleo_l152re/nucleo_l152re.dts [hunk 2]
```diff
@@ -39,6 +39,8 @@
 		led0 = &green_led_0;
 		sw0 = &user_button;
 		eeprom-0 = &eeprom;
+		adc1 = &adc1;
+		dac1 = &dac1;
```
Comment (gmarull) [inline boards/arm/nucleo_l152re/nucleo_l152re.dts:43]: I think this is redundant as sample does not force aliases. Could just use `DT_NODELABEL` as other boards do in the sample.

Comment (tagunil) [reply boards/arm/nucleo_l152re/nucleo_l152re.dts:43]: I see. Aliases were the old way of doing things. I'll clean this up for both Nucleo boards by a separate commit.

File: drivers/pinmux/stm32/pinmux_stm32l1x.h (hunk 1 above)
Comment (tagunil) [reply drivers/pinmux/stm32/pinmux_stm32l1x.h:217]: Looks like there are inconsistencies in pinmux ordering between STM32 series. I'll move these lines.

Comment (tagunil) [reply drivers/pinmux/stm32/pinmux_stm32l1x.h]: Done.

File: boards/arm/nucleo_l152re/nucleo_l152re.dts (hunk 2 above)
Comment (tagunil) [reply boards/arm/nucleo_l152re/nucleo_l152re.dts]: Done.

File: boards/arm/nucleo_l152re/pinmux.c [hunk 3]
```diff
@@ -24,6 +24,9 @@ static const struct pin_config pinconf[] = {
 #if DT_HAS_NODE_STATUS_OKAY(DT_NODELABEL(adc1))
 	{STM32_PIN_PA0, STM32L1X_PINMUX_FUNC_PA0_ADC1_IN0},
 #endif
+#if DT_HAS_NODE_STATUS_OKAY(DT_NODELABEL(dac1))
```
Comment (gmarull) [inline boards/arm/nucleo_l152re/pinmux.c:27]: Needs to be updated to `#if DT_NODE_HAS_STATUS(DT_NODELABEL(dac1), okay) && CONFIG_DAC`

Comment (tagunil) [reply boards/arm/nucleo_l152re/pinmux.c]: Done, thanks!

**Review by gmarull — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
@tagunil can you rebase once again? I'm not sure how to to make buildkite happy

**Pull request merged**